src/axom/sina/core/ID.cpp

```diff
@@ -65,10 +65,11 @@ IDField::IDField(ID value_, std::string localName_, std::string globalName_)
 IDField::IDField(conduit::Node const &parentObject,
                  std::string localName_,
                  std::string globalName_)
-  : IDField {extractIDFromObject(parentObject, localName_, globalName_),
-             std::move(localName_),
-             std::move(globalName_)}
-{ }
+  : value(extractIDFromObject(parentObject, localName_, globalName_))
+{
+  std::swap(localName, localName_);
+  std::swap(globalName, globalName_);
+}
```

Comment:
@doutriaux1 @bgunnar5 @gwaegner -- this was tricky to track down. 

Presumably msvc has a different order of processing the arguments than clang and other compilers, or it wiped out the strings after moving them, rather than leaving them unchanged.

The call to `extractIDFromObject` was happening after `localName_` and `globalName_` were already moved.